services/duty.service.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  validateDutyExists,
+  validateDutyIsModifiable,
+} from './validation/duty_validation';
+import {
+  Duty,
+  DutyQuery,
+  DutyQuerySchema,
+} from '../services/schemas/duty.zschema';
+import {
+  insertDuty,
+  getDutyByID,
+  getDutiesByQuery,
+  deleteDutyByID,
+  updateDutyByID,
+} from '../repositories/duty.repository';
+
+const createDuty = async (duty: Duty) => {
+  const result = await insertDuty({
+    ...duty,
+    soldiers: duty.soldiers ? duty.soldiers : [],
+  });
+  return result;
+};
+
+const findDutyByID = async (id: string) => {
+  const result = await getDutyByID(id);
+
+  validateDutyExists(id, result);
+
+  return result;
+};
+
+const findDutiesByQuery = async (query: DutyQuery) => {
+  const result = await getDutiesByQuery(query);
+  return result;
+};
+
+const deleteDuty = async (id: string) => {
+  const result = await getDutyByID(id);
+
+  validateDutyExists(id, result);
+
```

Comment:
No need to add a space between every function

Reply:
I think it looks cleaner than not adding a space

Reply:
ok

Reply:
resolved